homeassistant/components/homekit/accessories.py

```diff
@@ -137,6 +139,47 @@ def update_state(self, new_state):
         """
         raise NotImplementedError()
 
+    def sent_logbook_message(self, homekit_event, logbook_message):
+        """ Sent event to logbook."""
+
+        if logbook_message:
+            logbook_attr_message = "{} for {}".format(
+                logbook_message, self.display_name)
+        elif service:
+            logbook_attr_message = "{} for {}".format(
+                service, self.display_name)
+        else:
+            logbook_attr_message = "for {}".format(
+                self.display_name)
+
+        self.hass.bus.fire(homekit_event, {
+            ATTR_ENTITY_ID: self.entity_id,
+            'message': logbook_attr_message
+        })
+
+
+    def set_state(self, DOMAIN, service, params, logbook_message):
+        """ Sent event for logbook that a change was initiated from homekit.
+
+        Then call the service to change the state.
+        This method is to be called from methods that change states using
+        super().set_state(DOMAIN, service, params, logbook_message).
+        """
+
+        try:
+            if logbook_message:
+                message = "Send command {}".format(logbook_message)
+            else:
+                message = "Send command {}".format(service)
+
+            self.sent_logbook_message(EVENT_HOMEKIT_CHANGED, message)
+        except Exception as ex:
```

Comment:
Figure that firing the event itself results into going through some additional coding itself. My idea here for the try/except is to ensure that if something goes wrong/fails for the logbook that it does not interfere with the action itself.

For example, I trigger the event to lock my front door and for some reason (whatever it might be) the coding for the logbook (be it what we have or firing) has any kind of issue then the actual event to lock the door is not performed. By putting it within a try/except we make sure that the event to lock the door will still occur. 
More of a safety thing to ensure logging entries in logbook do not interfere with calling the action in case of failure.

We could call the service 1st, but then in the logbook it might show the action first before the entry from HomeKit which would be confusing.

Reply:
The beauty of the Home Assistant core being `async` is that one part doesn't interface directly with another. In this case:
 1. The HomeKit component will just ask the core to fire the event
 2. The core will then add jobs to do all tasks that listen for the event to the `event loop`
 3. The HomeKit method runs until completion
 4. The event loop executes previously added jobs

In summary, if something fails in the logbook component, it would **not** affect `HomeKit`. The `try` block isn't needed.

If you'r interested in async, this might be a good start: https://developers.home-assistant.io/docs/en/asyncio_101.html

Reply:
Thanks for the link, will definitely read on it. My main point was just if there was anything going wrong on the firing of the event itself then it would not stop changing the state in itself.
As I mentioned, a safety precaution in case something happens. 

Removed the try/except.